### PR TITLE
Favorite Pages: Notifying the user when a workflow action request fails

### LINKED ISSUE #25096: Favorite Pages: Notifying the user when a workflow action request fails
### User Story
As a dotCMS user, I want to receive a notification when a workflow action request fails, so I can take swift action. 

### Acceptance Criteria
- In the catchError of the workflow action trigger, show a toast with the error message.

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
N/A

### Quality Assurance Notes & Workarounds
1. Start dotCMS no license full starter
2. Go to pages
3. Trigger "publish" on a blog page
4. You should see a toast with the error

### Sub-Tasks & Estimates:
N/A

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Notifying the user when a workflow action request fails on page portlet

### Checklist
- [x] Tests
- [x] Translations

### <samp>🤖 Generated by Copilot at 6b655d2</samp>

Added error handling logic to dot-pages store when firing a workflow action to a page. Modified `dot-pages.store.ts` and `dot-pages.store.spec.ts` to implement and test the logic. Added `mockPublishAction` object to `dot-workflows-actions.mock.ts` to simulate a publish action in the test.

## Related Issue
Fixes #25096 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 6b655d2</samp>

*  Add error handling logic to dot-pages store when a workflow action request fails ([link](https://github.com/dotCMS/core/pull/25102/files?diff=unified&w=0#diff-1145b6e61f8f24833b827ae5f3fe939586964551b15aab913ba530cd44808f0eL791-R799))
*  Add test case to dot-pages store spec file to verify error handling logic ([link](https://github.com/dotCMS/core/pull/25102/files?diff=unified&w=0#diff-b23eeff307d6d254cb33a17b3b64f937646e6f91d72778c1ac460881c0bfc980R764-R793))
*  Add `mockPublishAction` object to dot-workflows-actions mock file ([link](https://github.com/dotCMS/core/pull/25102/files?diff=unified&w=0#diff-7b3c8f4445bdd8968d4c0c89da1fa32424154d6399b4b2f388fd1461613be98bR70-R87))

### Screenshots

https://github.com/dotCMS/core/assets/72418962/11b35adb-eca9-44e0-aa93-6ca633c7f415

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts [hunk 1]
```diff
@@ -1037,4 +1037,52 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
...
+        const isPage = item.baseType === 'HTMLPAGE';
+        const isContent = item.baseType === 'CONTENT';
+
+        // Page permissions
+        const canUserReadPage = isPage && loggedUser.canRead.htmlPages == true;
+        const canUserWritePage = isPage && loggedUser.canWrite.htmlPages == true;
+
+        // Contentlet permissions
+        const canUserReadContent = isContent && loggedUser.canRead.contentlets == true;
+        const canUserWriteContent = isContent && loggedUser.canWrite.contentlets == true;
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:1063]: Maybe the assertion with `true` is not needed

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:1063]: Lol, yeah. Just move the logic to a function. Forgot about that.

Comment (zJaaal) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:1063]: been there, done that

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts [hunk 2]
```diff
@@ -1037,4 +1037,52 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
...
+     * @param {DotCMSContentlet} item
+     * @return {*}  {UserPagePermission}
+     * @memberof DotPageStore
+     */
+    private getUserPagePermissions(item: DotCMSContentlet): UserPagePermission {
+        // Logged user
+        const { loggedUser } = this.get();
+
+        // Item types
+        const isPage = item.baseType === 'HTMLPAGE';
```
Comment (manuelrojas) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:1054]: @rjvelazco should we create a constant for `CONTENT` and `HTMLPAGE`?

Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts:1054]: Could be a good idea, idk.

Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts]: Done!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.spec.ts [hunk 3]
```diff
@@ -759,4 +761,34 @@ describe('DotPageStore', () => {
             inode: testInode
         });
     });
+
+    it('should handle error when a Workflow Action Request fails', (done) => {
+        const actions = [mockPublishAction];
+        const page = dotcmsContentletMock;
+        const error: HttpErrorResponse = new HttpErrorResponse({
+            error: {
+                message: 'Workflow Action is not available for current step'
```
Comment (manuelrojas) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.spec.ts:770]: I believe it should be: Workflow Action is not available for the current step.?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts]: we have `DotCMSBaseTypesContentTypes`

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.spec.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.spec.ts:770]: That's just a mock but I can change it

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts]: Done!

**Review by manuelrojas — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by danielmdob — APPROVED** [review]
LGTM!

**Review by fmontes — APPROVED** [review]

**Pull request merged**